test/e2e/healthcheck/kubernetes/virtual_probes.go

```diff
@@ -0,0 +1,81 @@
+package kubernetes
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/gruntwork-io/terratest/modules/k8s"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	config_core "github.com/kumahq/kuma/pkg/config/core"
+	. "github.com/kumahq/kuma/test/framework"
+	"github.com/kumahq/kuma/test/framework/deployments/testserver"
+)
+
+func VirtualProbes() {
+	namespaceWithSidecarInjection := func(namespace string) string {
+		return fmt.Sprintf(`
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: %s
+  annotations:
+    kuma.io/sidecar-injection: "enabled"
+`, namespace)
+	}
+
+	var k8sCluster Cluster
+	var optsKubernetes = KumaK8sDeployOpts
+
+	E2EBeforeSuite(func() {
+		k8sClusters, err := NewK8sClusters([]string{Kuma1}, Silent)
+		Expect(err).ToNot(HaveOccurred())
+
+		k8sCluster = k8sClusters.GetCluster(Kuma1)
+
+		Expect(Kuma(config_core.Standalone, optsKubernetes...)(k8sCluster)).To(Succeed())
+		Expect(YamlK8s(namespaceWithSidecarInjection(TestNamespace))(k8sCluster)).To(Succeed())
+		Expect(k8sCluster.VerifyKuma()).To(Succeed())
+	})
+
+	E2EAfterSuite(func() {
+		Expect(k8sCluster.DeleteNamespace(TestNamespace)).To(Succeed())
+		Expect(k8sCluster.DeleteKuma(optsKubernetes...)).To(Succeed())
+		Expect(k8sCluster.DismissCluster()).To(Succeed())
+	})
+
+	testServerReady := func() bool {
+		output, err := k8s.RunKubectlAndGetOutputE(k8sCluster.GetTesting(), k8sCluster.GetKubectlOptions(TestNamespace), "get", "pods")
+		if err != nil {
+			return false
+		}
+		lines := strings.Split(output, "\n")
+		if len(lines) != 2 {
+			return false
+		}
+		// 0:NAME 1:READY 2:STATUS 3:RESTARTS 4:AGE
+		words := strings.Fields(lines[1])
+		if len(words) != 5 {
+			return false
+		}
+		if words[1] == "2/2" && words[2] == "Running" && words[3] == "0" {
+			return true
+		}
+		return false
+	}
+
+	It("should deploy test-server with probes", func() {
+		Expect(testserver.Install(
+			testserver.WithArgs("--probes"),
+			testserver.WithHTTPProbes(),
+		)(k8sCluster)).To(Succeed())
+		Eventually(testServerReady, "5s", "100ms").Should(BeTrue())
```

Comment:
This is already checked in testserver.Install no?

Reply:
Can we include this check in `testserver.Install`? This can be the problem in every usage of TestServer.

The same way `VerifyKuma()` should be a part of `Kuma()` but that's another story.

Reply:
Agree. Do you think HTTP probes should be default for `test-server` deployment across all e2e tests?

Reply:
yes